docs/source/daml/reference/packages.rst

```diff
@@ -12,75 +12,94 @@ This page gives reference information on DAML package dependencies:
 
 .. code-block:: yaml
 
-  sdk-version: 0.0.0
-  name: foo
-  source: daml
-  version: 1.0.0
-  exposed-modules:
-    - Some.Module
-    - Some.Other.Module
   dependencies:
     - daml-prim
     - daml-stdlib
-    - /home/johndoe/bar/.daml/dist/bar-1.0.0.dar
+    - ../foo/foo.dar
 
-The import path needs to be the relative or absolute path pointing to the created DAML archive of
-the ``bar`` project. The archive can reside anywhere on the local file system. Note that the SDK
-versions of the packages ``foo`` and ``bar`` need to match, i.e. it is an error to import a package
-that was created with an older SDK.
+The import path can also be absolute, for example, by changing the last line to:
 
-Once a package has been added to the dependencies of the ``foo`` project, modules of ``bar`` can be
-imported as usual with the ``import Some.Module`` directive (see :ref:`Imports <daml-ref-imports>`).
-If both projects ``foo`` as well as ``bar`` contain a module with the same name, the import can be
-disambiguated by adding the package name in front of the module name, e.g. ``import "bar"
-Some.Module``.
+.. code-block:: yaml
 
-Note that all modules of package ``foo`` that should be available as imports of other packages need
-to be exposed by adding them to the ``exposed-modules`` stanza of the `daml.yaml` file. If the
-``exposed-modules`` stanza is omitted, all modules of the project are exposed by default.
+    - /home/user/foo/foo.dar
 
-Importing archives compiled with different SDK's
-************************************************
+When you run ``daml build`` in ``bar`` project, the compiler will make the definitions in ``foo.dar`` available for importing. For example, if ``foo`` exports the module ``Foo``, you can import it in the usual way:
 
-All DAML archive dependencies of a project need to be compiled with the same SDK as the project
-itself. However, it is possible to import templates and data types of an archive compiled with an
-older SDK, by listing them under the ``data-dependencies`` stanza:
+.. code-block:: daml
 
-.. code-block:: yaml
+  import Foo
+
+Sometimes you will have multiple packages with the same exported module name. In that case, you must also specify which package the module comes from, as follows:
+
+.. code-block:: daml
+
+  import "foo" Foo
+
+By default, all modules of ``foo`` are made available when importing ``foo`` as a dependency. To control which modules of ``foo`` get exported, you may add an ``exposed-modules`` field in the ``daml.yaml`` file for ``foo``:
 
-  sdk-version: 0.0.0
-  name: foo
-  source: daml
-  version: 1.0.0
   exposed-modules:
-    - Some.Module
-    - Some.Other.Module
+  - Foo
+
+**Important Limitation:** For DAML archive imports via ``dependencies`` to work, the archive should be compiled with the same DAML SDK version. Otherwise, compilation will fail. If matching the SDK version is not possible, see the next section.
```

Comment:
I expect that at least for now where libraries don’t really exist, most users will actually want `data-dependencies` not `dependencies` so the warning here at the end might be a bit late. What do you think of adding a small section at the beginning that compares them and maybe swapping the order of `dependencies` and `data-dependencies`?

Happy to leave that out of this PR.